src/sdl_font.cpp

```diff
@@ -323,30 +187,82 @@ CachedTTFFont::CachedTTFFont(
-        dbg( D_INFO ) << "Using compatible font [" + typeface + "] found in font dir.";
-    }
-    //different default font with wincurse
-    if( !file_exist( typeface ) ) {
-        faceIndex = 0;
-        typeface = PATH_INFO::fontdir() + "unifont.ttf";
-        dbg( D_INFO ) << "Using fallback font [" + typeface + "] found in font dir.";
-    }
-    dbg( D_INFO ) << "Loading truetype font [" + typeface + "].";
+    std::vector<std::string> typefaces;
+    std::vector<std::string> known_suffixes = { ".ttf", ".fon" };
+    bool add_suffix = true;
+    for( const std::string &ks : known_suffixes ) {
+        if( string_ends_with( typeface, ks ) ) {
+            add_suffix = false;
+            break;
+        }
+    }
+    bool add_prefix = true;
+    std::vector<std::string> known_prefixes = {
+        PATH_INFO::user_font(), PATH_INFO::fontdir()
+    };
+
+#if defined(_WIN32)
+    constexpr UINT max_dir_len = 256;
+    char buf[max_dir_len];
+    const UINT dir_len = GetSystemWindowsDirectory( buf, max_dir_len );
+    if( dir_len == 0 ) {
+        throw std::runtime_error( "GetSystemWindowsDirectory failed" );
+    } else if( dir_len >= max_dir_len ) {
+        throw std::length_error( "GetSystemWindowsDirectory failed due to insufficient buffer" );
+    }
+    known_prefixes.emplace_back( buf + std::string( "\\fonts\\" ) );
+#elif defined(_APPLE_) && defined(_MACH_)
+    /*
+    // Well I don't know how osx actually works ....
+    known_prefixes.emplace_back( "/System/Library/Fonts/" );
+    known_prefixes.emplace_back( "/Library/Fonts/" );
+    wordexp_t exp;
+    wordexp( "~/Library/Fonts/", &exp, 0 );
+    known_prefixes.emplace_back( exp.we_wordv[0] );
+    wordfree( &exp );
+    */
+#else // Other POSIX-ish systems
+    known_prefixes.emplace_back( "/usr/share/fonts/" );
+    known_prefixes.emplace_back( "/usr/local/share/fonts/" );
+    char *home;
+    if( ( home = getenv( "HOME" ) ) ) {
+        std::string userfontdir = home;
+        userfontdir += "/.fonts/";
+        known_prefixes.emplace_back( userfontdir );
+    }
+#endif
+
+    for( const std::string &kp : known_prefixes ) {
+        if( string_starts_with( typeface, kp ) ) {
+            add_prefix = false;
+            break;
+        }
+    }
+
+    for( const std::string &ks : known_suffixes ) {
+        for( const std::string &kp : known_prefixes ) {
+            if( add_prefix ) {
+                typefaces.emplace_back( kp + typeface + ( add_suffix ? ks : "" ) );
+            }
+            typefaces.emplace_back( typeface + ( add_suffix ? ks : "" ) );
+        }
+    }
+    if( typefaces.empty() ) {
+        typefaces.emplace_back( typeface );
+    }
+    ensure_unifont_loaded( typefaces );
+
+    for( const std::string &tf : typefaces ) {
+        if( !file_exist( tf ) ) {
+            dbg( D_ERROR ) << "Not found " << tf;
+            continue;
+        }
+        dbg( D_ERROR ) << "Loading truetype font " << tf;
```

Comment:
`D_ERROR` makes the game emit a backtrace here. Was it supposed to be `D_INFO` or something?

Reply:
> 
> 
> `D_ERROR` makes the game emit a backtrace here. Was it supposed to be `D_INFO` or something?

Yes, log level should've been reduced.